pkg/storagev2/leveldb/store.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package leveldbstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	storageV2 "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/hashicorp/go-multierror"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/errors"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/storage"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type Store struct {
+	DB      *leveldb.DB
+	path    string
+	closeLk sync.RWMutex
+}
+
+var _ storageV2.Store = (*Store)(nil)
+var _ storageV2.Tx = (*transaction)(nil)
```

Comment:
Let's remove the transaction part from here. We should do it separately. I am not sure yet if the transaction will only be concerned about levelDB.